packages/cli/cloud/src/environment/list/action.ts

```diff
@@ -0,0 +1,55 @@
+import chalk from 'chalk';
+import type { CLIContext } from '../../types';
+import { cloudApiFactory, local, tokenServiceFactory } from '../../services';
+import { promptLogin } from '../../login/action';
+
+async function getProject(ctx: CLIContext) {
+  const { project } = await local.retrieve();
+  if (!project) {
+    ctx.logger.warn(
+      `\nWe couldn't find a valid local project config.\nPlease link your local project to an existing Strapi Cloud project using the ${chalk.cyan(
+        'link'
+      )} command`
+    );
+    process.exit(1);
+  }
+  return project;
+}
+
+export default async (ctx: CLIContext) => {
+  const { getValidToken } = await tokenServiceFactory(ctx);
+  const token = await getValidToken(ctx, promptLogin);
+  const { logger } = ctx;
+
+  if (!token) {
+    return;
+  }
+
+  const project = await getProject(ctx);
+  if (!project) {
+    return;
```

Comment:
WDYT of adding a debug log?

Reply:
We have a warning log in the getProject. This is a 'safety net' in case there is no project even after the getProject() function run (which seems very unlikely). A debug log would be correct but maybe it would not add much value in this case. Same for the token one mentioned in the comment above. Do you think that we should still add it? If we add it, we should also do it for the other command's actions since we're following the same pattern.

Reply:
It makes sense to add logs in unusual end-of functions, but in this case, I think the log created in getProject() function already serves that purpose. I would say we either:

1- Discard the redundant check
2- Add a log before returning

For any of both decisions, we should refactor all the functions that follow the same pattern.